### PR TITLE
[IOPAY-402] update node 14 & fp-ts

### PR DESCRIPTION
**Pull request by infantesimone**
#### List of Changes

- update `node 14.16.0`
- update `io-ts  2.2.16`
- update `fp-ts 2.11.1`
- update `jest 27`
- migrate `tslint` --> `eslint`
- refactoring according to `fp-ts 2.11.1`
  
#### Motivation and Context

The goal of this PR is to update some critical dependences, before porting the `pagopa-proxy` from the `PROD-IO` to the `pagoPA`Azure subscription (according to [this RFC](https://pagopa.atlassian.net/wiki/spaces/I/pages/440632330/RFC+-+porting+pagopa-proxy+in+pagoPA+draft)).

#### How Has This Been Tested?

- unit test or run postman collection locally.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by infantesimone** [comment]
@giovanniberti the goal is to run pipeline in `pagopa-projects`, so according to the following configuration:
https://github.com/pagopa/gitops/blob/391fba67fe8b4963e8e91e801c09679adf81049f/azure-devops/projects/pagopa-projects/pagopa-proxy.tf#L8

you have to change the yaml prefix with pagopa, so:
- pagopa-code-review-pipelines.yml
- pagopa-deploy-pipelines.yml

File: .devops/deploy-pipelines.yml [hunk 1]
```diff
@@ -154,7 +154,40 @@ stages:
...
+      - Deploy_test
+    jobs:
+      - job: 'TestAPI'
+        steps:
+          - checkout: none
+
+          - download: current
+            artifact: Bundle
+    
+      - job: waitStartup
```
Comment (pasqualespica) [inline .devops/deploy-pipelines.yml:169]: is this necessary to wait proxy get up?

File: package.json [hunk 2]
```diff
@@ -22,24 +22,25 @@
...
     "generate:ts-from-yaml:nodeNm3io": "shx mkdir -p generated/nodeNm3io && rimraf generated/nodeNm3io/* && gen-api-models --strict 0 --api-spec generated/yaml-file/nodeForIO_Service.yaml --out-dir generated/nodeNm3io",
     "generate:yaml-from-wsdl": "shx mkdir -p generated/yaml-file && rimraf generated/yaml-file/* && node pagopa_api/YamlGenerator.js",
-    "generate:proxy-api-models": "shx mkdir -p generated/api && rimraf generated/api/* && gen-api-models --api-spec ./api_pagopa.yaml --out-dir generated/api --ts-spec-file generated/api/public_api_pagopa.ts --request-types",
+    "generate:proxy-api-models": "shx mkdir -p generated/api && rimraf generated/api/* && gen-api-models --api-spec ./api_pagopa.yaml --out-dir generated/api --request-types",
     "prettify": "prettier --write \"./**/*.ts\"",
     "start": "node dist/src/index.js",
-    "test:coverage": "jest --runInBand --coverage",
-    "test:integration": "jest --config=jest.integration.js --runInBand",
+    "test:coverage": "jest --runInBand --coverage --detectOpenHandles",
```
Comment (pasqualespica) [inline package.json:28]: with `--detectOpenHandles` option , when i runs  test i retrieve the following nasty warning message
```sh
Jest has detected the following 1 open handle potentially keeping Jest from exiting:

  ●  TCPWRAP
```

is it correct to display it? or is there a way to disable it?

File: .devops/deploy-pipelines.yml [hunk 3]
```diff
@@ -154,7 +154,40 @@ stages:
...
+          - script: |
+              yarn global add newman
+            displayName: 'newman installation'
+
+          - task: Bash@3
+            inputs:
+              targetType: filePath
+              filePath: k6example/ws.sh
+              arguments: postman
+              workingDirectory: k6example
```
Comment (pasqualespica) [inline .devops/deploy-pipelines.yml:187]: could  be better to rename `k6example` folder in to `api-test`

**Review by pasqualespica — APPROVED** [review]
great work 🚀

**Comment by gunzip** [comment]
Very good and really important upgrade! thank you @infantesimone!

File: .devops/deploy-pipelines.yml (hunk 1 above)
Comment (infantesimone) [reply .devops/deploy-pipelines.yml:169]: you're right, thanks

File: package.json (hunk 2 above)
Comment (infantesimone) [reply package.json:28]: added for some debugging, but we can disable, thanks

**Review by giovanniberti — APPROVED** [review]

**Comment by infantesimone** [comment]
In order to leave the master branch aligned with the `pagopa-proxy` in `PROD-IO`, this PR is merged into a new release branch in which all porting activities will be integrated.

**Pull request merged**